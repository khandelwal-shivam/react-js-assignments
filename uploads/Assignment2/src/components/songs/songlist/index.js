import "./songlist.css";
import { SongData } from "../songdata";

const SongList = (props) => {
  return (
    <section className="song-list">
      <h2 className="txt-center">Songs List</h2>
      <table className="playlist">
        <thead>
          <tr>
            <th>Movie</th>
            <th>Title</th>
            <th>Song Length</th>
            <th>Singer</th>
          </tr>
        </thead>
        <tbody>
          {SongData.map((element, index) => {
            return (
              <tr key={index}>
                <td>{element.movie}</td>
                <td>{element.title}</td>
                <td>{element.length}</td>
                <td>{element.singer}</td>
              </tr>
            );
          })}
        </tbody>
      </table>
    </section>
  );
};
export default SongList;
