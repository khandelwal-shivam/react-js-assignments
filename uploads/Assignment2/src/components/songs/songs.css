.info * {
  margin: 0.5rem 0;
}

/* Add song Form */

.song-form {
  width: 500px;
  height: 500px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.song-form text{
  color: red;
}
.input-group {
  margin: 0.5rem 0;
  width: inherit;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.input-group input {
  font-size: 1rem;
  padding-left: 0.5rem;
  margin: 0 0.5rem;
  width: 15rem;
  height: 2rem;
}

#btn-addSong {
  margin: 1rem 2rem;
  border: 1px solid green;
  width: 100%;
  height: 3rem;
  font-size: 1.25rem;
}

#btn-addSong:hover {
  border: initial;
  background: green;
}
