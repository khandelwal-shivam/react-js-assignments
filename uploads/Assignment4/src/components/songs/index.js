import { Switch, Route, useHistory } from "react-router-dom";

import SongList from "./songlist";
import SongForm from "./songform";
import SongInfo from "./songInfo";
const AllSongs = (props) => {
  const history = useHistory();
  // console.log("history",history);
  return (
    <Switch>
      <Route exact path="/songs" component={SongList} />
      <Route exact path="/songs/addsong" component={SongForm} />
      <Route path="/songs/:id" component={SongInfo} />
    </Switch>
  );
};
export default AllSongs;
