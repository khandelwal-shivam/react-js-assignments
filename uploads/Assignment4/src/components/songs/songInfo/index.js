import { Fragment } from "react";
import { useState, useEffect } from "react";
import { useHistory } from "react-router";
import { useParams } from "react-router-dom";

import "./songInfo.css";

const SongInfo = (props) => {
  const history = useHistory();
  let { id } = useParams();
  const [song, setSong] = useState({});

  const fetchSongs = async (id) => {
    console.log("id: ", id);
    const response = await fetch(`http://localhost:30001/songlist/${id}`);
    const data = await response.json();
    setSong(data);
  };

  useEffect(() => {
    fetchSongs(id);
  }, []);

  return (
    <Fragment>
      <section className="song-detail">
        <div className="song-info">
          <table>
            <thead>
              <tr>
                <th>Song Id</th>
                <th>Movie</th>
                <th>Song Title</th>
                <th>Song Length</th>
                <th>Singer</th>
              </tr>
            </thead>

            <tbody>
              <tr>
                <td>{song["id"]}</td>
                <td>{song["movie"]}</td>
                <td>{song["title"]}</td>
                <td>{song["length"]}</td>
                <td>{song["singer"]}</td>
              </tr>
            </tbody>
          </table>
          <button
            className="btn back-btn"
            onClick={() => {
              history.push("/songs");
            }}
          >
            Back
          </button>
        </div>
      </section>
    </Fragment>
  );
};
export default SongInfo;
