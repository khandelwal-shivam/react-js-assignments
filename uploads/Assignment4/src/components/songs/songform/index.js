import "./songform.css";
import { useFormik } from "formik";
import Axios from "axios";
import { v4 as uuid } from "uuid";
import { useHistory } from "react-router-dom";
import { Grid } from '@material-ui/core';
import * as Yup from 'yup';
const SongForm = () => {
  const history = useHistory();
  
  const validationSchema = Yup.object({
    movie: Yup.string().required("Required!!!"),
    title: Yup.string().required("Required!!!"),
    length: Yup.string().required("Required!!!"),
    singer: Yup.string().required("Required!!!")
  });

  const initialValues = {
    movie: "",
    title: "",
    length: "",
    singer: "",
  }
  const onSubmit =  (values) => {
    console.log("values from formik ",values);
    values["id"] = uuid();
    Axios.post("http://localhost:30001/songs", values)
      .then((resp) => {
        history.push("/songs");
      })
      .catch((error) => {
        console.log(error);
      });
  }

  const formik = useFormik({
    initialValues,
    onSubmit,
    validationSchema
  });
  console.log("touched",formik.touched);
  console.log("errors",formik.errors);
  return (
    <section className="song-form">
      <form className="form" onSubmit={formik.handleSubmit}>
        <Grid container className="input-group" >
          <Grid item xs = {3}><label htmlFor="movie">Movie</label></Grid>
          <Grid item xs = {5}>
            <input
            id="movie"
            name="movie"
            type="text"
            onChange={formik.handleChange}
            value={formik.values.movie}
            onBlur = {formik.handleBlur}
          /></Grid>
          <Grid item xs = {4}>
          {formik.errors.movie && formik.touched.movie ? (
            <div className="validation-error flex">{formik.errors.movie}</div>
          ) : null}
          </Grid>
        </Grid>
        <Grid container className="input-group" >
          <Grid item xs = {3}><label htmlFor="title">Song Title</label></Grid>
          <Grid item xs = {5}>
            <input
            id="title"
            name="title"
            type="text"
            onChange={formik.handleChange}
            value={formik.values.title}
            onBlur = {formik.handleBlur}
          /></Grid>
          <Grid item xs = {4}>
          {formik.errors.title && formik.touched.title ? (
            <div className="validation-error flex">{formik.errors.title}</div>
          ) : null}
          </Grid>
        </Grid>
        <Grid container  className="input-group">
          <Grid item xs = {3}><label htmlFor="length">Song length</label></Grid>
          <Grid item xs = {5}>
            <input
            id="length"
            name="length"
            type="text"
            onChange={formik.handleChange}
            value={formik.values.length}
            onBlur = {formik.handleBlur}
          /></Grid>
          <Grid item xs = {4}>
          {formik.errors.length && formik.touched.length ? (
            <div className="validation-error flex">{formik.errors.length}</div>
          ) : null}
          </Grid>
        </Grid>
        <Grid container  className="input-group">
          <Grid item xs = {3}><label htmlFor="singe">Singer</label></Grid>
          <Grid item xs = {5}>
            <input
            id="singer"
            name="singer"
            type="text"
            onChange={formik.handleChange}
            value={formik.values.singer}
            onBlur = {formik.handleBlur}
          /></Grid>
          <Grid item xs = {4}>
          {formik.errors.singer && formik.touched.singer ? (
            <div className="validation-error flex">{formik.errors.singer}</div>
          ) : null}
          </Grid>
        </Grid>

        <button className="btn submit-btn" type="submit">
          Submit
        </button>
      </form>
    </section>
  );
};

export default SongForm;
