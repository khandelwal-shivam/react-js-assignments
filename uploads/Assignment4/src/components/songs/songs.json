{
  "songs": [
    {
      "id": 1,
      "movie": "Anurodh",
      "title": "Jab Dard Nahi Tha",
      "length": "04:50",
      "singer": "Kishore"
    },
    {
      "id": 2,
      "movie": "Kabhi Kabhi",
      "title": "Me Pal Do Pal",
      "length": "03:45",
      "singer": "Mukesh"
    },
    {
      "id": 3,
      "movie": "Aradhana",
      "title": "Aaj Tujhko",
      "length": "04:45",
      "singer": "Rafi"
    },
    {
      "id": 4,
      "movie": "Mungaru Male",
      "title": "Anisuthidhe",
      "length": "05:45",
      "singer": "Sonu Nigam"
    }
  ]
}
