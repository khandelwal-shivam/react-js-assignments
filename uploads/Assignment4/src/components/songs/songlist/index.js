import "./songlist.css";
import { useSelector,useDispatch } from "react-redux";
import { useState, useEffect } from "react";
import { fetchSongs } from "../../../Redux/Actions";
import { useHistory } from "react-router-dom";
const SongList = () => {
  // const [songData, setSongData] = useState([]);
  const songData =  useSelector(state=>state.songs.songList);
  const dispatch = useDispatch();
  const history = useHistory();
  // const fetchSongs = async () => {
  //   const response = await fetch("http://localhost:30001/songs");
  //   const data = await response.json();
  //   setSongData(data);
  // };

  useEffect(() => {
    dispatch(fetchSongs());
  }, [dispatch]);

  return (
    <section className="song-list">
      <h2>Song List</h2>
      <table className="playlist">
        <thead>
          <tr>
            <th>Movie</th>
            <th>Title</th>
            <th>Song Length</th>
            <th>Singer</th>
          </tr>
        </thead>
        <tbody>
          {songData && songData.map((element, index) => {
            return (
              <tr key={index}>
                <td>{element.movie}</td>
                <td>{element.title}</td>
                <td>{element.length}</td>
                <td>{element.singer}</td>
              </tr>
            );
          })}
        </tbody>
      </table>
      <button className = "addSong"
       onClick = {()=>{history.push('/songs/addsong')}}
      >Add Songs</button>
    </section>
  );
};
export default SongList;
