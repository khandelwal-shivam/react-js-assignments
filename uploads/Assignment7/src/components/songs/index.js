/* eslint-disable no-lone-blocks */
import { Fragment, useState } from "react";
import "./songs.css";

import SongList from "./songlist";
import { SongData, SongTemplate } from "./songdata";

const AllSongs = (props) => {
  const [songCount, setSongCount] = useState(SongData.length);
  var song = SongTemplate;

  const AddSong = (info) => {
    var data = {};
    var i = 0;
    for (const [key] of Object.entries(song)) {
      data[key] = info[i++].value;
    }
    console.log(data);
    song = data;
    SongData.push(song);
    setSongCount(songCount + 1);
  };

  {
    /* 
  const onFormSubmit = (event) => {
    event.preventDefault();
    AddSong(event.target);
    console.log("New Song Added to PlayList");
    return false;
  };

  */
  }
  return (
    <Fragment>
      <section className="info">
        <h2>Display Song Info using in memory JSON</h2>
        <h4 className="txt-center">
          No. of songs in the playlist : {songCount}
        </h4>
        <hr />
      </section>
      <SongList data={SongData} />
      {/*
        <section>
        <form className="song-form" onSubmit={onFormSubmit}>
          <h1 className="mb-1">Song Form</h1>
          <div className="input-group">
            <label htmlFor="movie">Movie Name</label>
            <input type="text" id="movie" required />
          </div>
          <div className="input-group">
            <label htmlFor="title">Song Title</label>
            <input type="text" id="title" required />
          </div>
          <div className="input-group">
            <label htmlFor="length">Song Length</label>
            <input type="text" id="length" required />
          </div>
          <div className="input-group">
            <label htmlFor="singer">Singer</label>
            <input type="text" id="singer" required />
          </div>
          <button className="btn" id="btn-addSong">
            Add Song
          </button>
        </form>
      </section>
        */}
    </Fragment>
  );
};
export default AllSongs;
