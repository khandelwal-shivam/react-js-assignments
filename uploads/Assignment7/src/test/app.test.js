import toJson from "enzyme-to-json";
import { shallow, mount } from "enzyme";
import Header from "../components/header";
import AllSongs from "../components/songs";
import { SongData } from "../components/songs/songdata";
import SongList from "../components/songs/songlist";
import Enzyme from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";

Enzyme.configure({ adapter: new Adapter() });

const traverseComponent = (component, index) => {
  let result;
  let cur = 0;
  component.forEach((node) => {
    if (cur === index) {
      result = node;
    }
    cur++;
  });
  return result;
};

describe("Test Suite 1", () => {
  it("SnapShot Testing for SongList using Mount", () => {
    const wrap = mount(<AllSongs />);
    expect(wrap).toMatchSnapshot();
  });

  it("Check if correct h1  heading name is rendered", () => {
    const actualHeading = "FED React : Assigment7 : TDD, JEST and Enzyme";
    const wrapper = shallow(<Header />);
    expect(wrapper.find("h1").text()).toEqual(actualHeading);
  });

  it("SongList Component is rendered", () => {
    const wrapper = shallow(<AllSongs />);
    expect(wrapper.find("SongList").length).toEqual(1);
  });
});

describe("Test Suite 2", () => {
  it("Title is 2nd column in SongList", () => {
    const actualHeader2 = "Title";
    const wrapper = shallow(<SongList data={SongData} />);
    const innerComp = traverseComponent(
      wrapper.find("thead tr th").children(),
      1
    );
    let expectedHeader2 = innerComp.text();
    expect(expectedHeader2).toEqual(actualHeader2);
  });

  it("Singer is 4th column heading in SongList", () => {
    const actualHeader4 = "Singer";
    const wrapper = shallow(<SongList data={SongData} />);
    const innerComp = traverseComponent(
      wrapper.find("thead tr th").children(),
      3
    );
    let expectedHeader4 = innerComp.text();
    expect(expectedHeader4).toEqual(actualHeader4);
  });

  it("2nd song name passed as props match with SongList Component as 2nd songName when rendered.", () => {
    const expectedSongName2 = SongData[1].title;
    let actualSongName2;
    let i = 0;
    const wrapper = mount(<SongList data={SongData} />);
    const components = wrapper.find(".playlist tbody tr").children();
    components.forEach((node) => {
      if (i === 5) {
        actualSongName2 = node.render().text();
      }
      i++;
    });
    expect(actualSongName2).toEqual(expectedSongName2);
  });

  it("When SongList Array passed as props then Second Song's SongLength matches with Second Song Component's song Length when rendered", () => {
    const ExpectedSongData = Object.values(SongData[1]);
    let ActualSongData = [];
    const wrapper = shallow(<SongList data={SongData} />);
    const SongListWrapper = wrapper.find(".playlist tbody").children();
    const SongInfo2 = traverseComponent(SongListWrapper, 1).find("td");

    SongInfo2.forEach((node) => {
      ActualSongData.push(node.render().text());
    });
    for (let i = 0; i < 4; i++) {
      expect(ActualSongData[i]).toEqual(ExpectedSongData[i]);
    }
  });

  it("When Null Array pass as SongList's props then it must not crash and no song component is rendered", () => {
    const wrapper = shallow(<SongList data={[]} />);
    expect(wrapper.find(".playlist tbody").children().length).toEqual(0);
  });
});
