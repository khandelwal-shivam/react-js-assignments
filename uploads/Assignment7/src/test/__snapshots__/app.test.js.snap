// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Suite 1 SnapShot Testing for SongList using Mount 1`] = `ReactWrapper {}`;
